asmde/allocator.py

```diff
@@ -361,7 +361,8 @@ def empty(self):
     def fallback(self):
         """ basic block fall backs to the next one if it does not end with
             a non-conditional jump """
-        return (not self.bundle_list[-1].has_nocond_jump) or self.bundle_list[-1].has_cond_jump
+        return not self.bundle_list or \
+                (not self.bundle_list[-1].has_nocond_jump) or self.bundle_list[-1].has_cond_jump
```

Comment:
Beware than in asmde "bundle" may be a generic term.
Every arch is expected to have bundles.
Some archs (e.g. VLIW) will have bundle with multiple instructions (and bundle separators) and other arch may have only single instruction bundles.
So some of the bugs you have encountered (such as not supporting consecutive labels ...) should not necessarily fixed this way.

Reply:
`hasBundle` is used to distinguish between arch with multi-instructions bundle (True) and arch with single instruction bundles (False).
I agree this can be confusing.

Reply:
I think I agree with all the other fixes, they seem to be a right use of `hasBundle`.
I wonder if this one is not hidding another issue (empty BB for example)

Reply:
Right, this change didn't make sense. I removed it.